tooling/minitest/src/tests/builder_api.rs

```diff
@@ -0,0 +1,115 @@
+use crate::*;
+
+#[test]
+fn global_var() {
+    let mut p = ProgramBuilder::new();
+    let var = p.global_int::<u32>();
+    let mut f = p.function();
+    f.assign(var, const_int(42u32));
+    f.if_(eq(load(var), const_int(42u32)), |_| {}, |f| f.unreachable());
+    f.exit();
+    p.finish_function(f);
+    let p = p.finish_program();
+    assert_stop(p);
+}
+
+#[test]
+fn local_var() {
+    let mut p = ProgramBuilder::new();
+    let mut f = p.function();
+    let var = f.local::<u32>();
```

Comment:
```suggestion
    let mut f = p.declare_function();
    let var = f.declare_local::<u32>();
```
I think it'd be good to have some pattern for all the functions that declare a fresh name. `f.ret()` could also mean "return from the function", after all.